bundles/org.openhab.binding.wizlighting/src/main/java/org/openhab/binding/wizlighting/internal/handler/WizLightingMediatorImpl.java

```diff
@@ -0,0 +1,252 @@
+                if (discoveryServe != null) {
+                    discoveryServe.discoveredLight(bulbMac, bulbIp);
+                    logger.trace("Sending a new thing to the discovery service.  MAC: {}  IP: {}", bulbMac, bulbIp);
+                } else {
+                    logger.trace("There is no discovery service registered to receive the new bulb!");
+                }
+            }
+        } else {
+            logger.warn("The sync response did not contain a valid mac address, it cannot be processed.");
+        }
+    }
+
+    /**
+     * Register one new {@link Thing} and the corresponding
+     * {@link WizLightingHandler}.
+     *
+     * @param thing the {@link Thing}.
+     * @param handler the {@link WizLightingHandler}.
+     */
+    @Override
+    public void registerThingAndWizBulbHandler(final Thing thing, final WizLightingHandler handler) {
+        this.handlersRegisteredByThing.put(thing, handler);
+    }
+
+    /**
+     * Unregister one {@link WizLightingHandler} by the corresponding {@link Thing}.
+     *
+     * @param thing the {@link Thing}.
+     */
+    @Override
+    public void unregisterWizBulbHandlerByThing(final Thing thing) {
+        this.handlersRegisteredByThing.remove(thing);
+    }
+
+    /**
+     * Utility method to get the registered thing handler in mediator by the mac
+     * address.
+     *
+     * @param bulbMacAddress the mac address of the thing of the handler.
+     * @return {@link WizLightingHandler} if found.
+     */
+    private @Nullable WizLightingHandler getHandlerRegisteredByMac(final String bulbMacAddress) {
+        WizLightingHandler searchedHandler = null;
+        for (WizLightingHandler handler : this.handlersRegisteredByThing.values()) {
+            if (bulbMacAddress.equalsIgnoreCase(handler.getBulbMacAddress())) {
+                searchedHandler = handler;
+                // don't spend more computation. Found the handler.
+                break;
+            }
+        }
+        return searchedHandler;
+    }
+
+    /**
+     * Inits the mediator WizBulbUpdateReceiverRunnable thread. This thread is
+     * responsible to receive all packets from Wiz Bulbs, and redirect the messages
+     * to mediator.
+     */
+    private void initMediatorWizBulbUpdateReceiverRunnable() {
+        WizLightingUpdateReceiverRunnable receiver = this.receiver;
+        Thread receiverThread = this.receiverThread;
+        // try with handler port if is null
+        if ((receiver == null)
+                || ((receiverThread != null) && (receiverThread.isInterrupted() || !receiverThread.isAlive()))) {
+            try {
+                logger.trace("Receiver thread is either null, interrupted, or dead.");
+                WizLightingUpdateReceiverRunnable newReceiver = new WizLightingUpdateReceiverRunnable(this,
+                        WizLightingBindingConstants.DEFAULT_LISTENER_UDP_PORT);
+                Thread newThread = new Thread(newReceiver);
```

Comment:
Please give the thread a name to identify that it came from this binding. Also make this thread daemon.

Reply:
Done